enumerable.rb

```diff
@@ -0,0 +1,186 @@
+module Enumerable
+  def my_each
+    if self.class == Range
+      l = to_a
+      count = 0
+      while count < l.length
+        yield(l[count])
+        count += 1
+      end
+      l
+    elsif self.class == Hash
+      count = 0
+      while count < length
+        yield(self[count])
+        count += 1
+      end
+      self
+    else
+      count = 0
+      while count < length
+        yield(self[count])
+        count += 1
+      end
+      self
+    end
+  end
+
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+
+  def my_select
+    array = []
+    my_each do |i|
+      array.push(i) if yield(i)
+    end
+    array
+  end
+
+  def my_all?
```

Comment:
* [ ] should have working `my_all?` **(covered by automated tests)** identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

``` p %w[ant bear cat].my_all?(/t/)  !=  %w[ant bear cat].all?(/t/)  ```
``` p [1, 2i, 3.14].my_all?(Numeric) != [1, 2i, 3.14].my_all?(Numeric)  ```
``` p [nil, true, 99].my_all?  != [nil, true, 99].my_all? ```